File: Lecture 21 Solving LeetCodeCodeStudio Questions [Arrays]/Rotate an Array by K.cpp
#include <iostream>
#include<bits/stdc++.h>

using namespace std;
//Program to rotate an array by K
void rotate(vector<int>& nums, int k) 
{
    vector<int> arr(nums.size());
    
    for(int i=0; i<nums.size(); i++)
    {
        arr[(i+k)%nums.size()] = nums[i];
    }
    
    nums = arr;
}
    
int main()
{
    vector<int> arr {1,2,3,4,5,6,7};
    int k = 3;
    rotate(arr,k);
	
	for(int i=0; i<arr.size(); i++)
	{
	    cout<<arr[i]<<" ";
	}
}

File: Lecture 21 Solving LeetCodeCodeStudio Questions [Arrays]/Rotate an Array by K.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.*;
public class Main{
//Program to rotate an array by K
public static void rotate(int []nums, int k) 
{
    int []arr = new int [nums.length];
    
    for(int i=0; i<nums.length; i++)
    {
        int g = (i+k)%nums.length;
        arr[g] = nums[i];
    }
    
    int f = 0;
    for(int t : arr)
    {
        nums[f] = t;
        f++;
    }
}
    
	public static void main(String[] args) 
	{
	    int arr[] = {1,2,3,4,5,6,7};
	    int k = 3;
	    rotate(arr,k);
    	for(int i=0; i<arr.length; i++)
    	{
    	    System.out.println(arr[i]);
    	}
	}
}
